client_new/src/types/snapshot.rs

```diff
@@ -129,6 +119,12 @@ impl SnapshotPath {
     }
 }
 
+impl Display for SnapshotPath {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "SnapshotPath: {:?}", self.path)
+    }
+}
+
```

Comment:
Nit: This is currently basically identical to the `Debug` impl. When printing an error message that includes a `SnapshotPath`, the reader probably doesn't care that this is a `SnapshotPath`, the actual inner path would be sufficient. For `Debug` context is good, for `Display` it's not necessary.

```suggestion
impl Display for SnapshotPath {
    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
        write!(f, "{}", self.path.display())
    }
}

```